dds/DCPS/RTPS/Sedp.h

```diff
@@ -378,25 +387,22 @@ class Sedp : public DCPS::EndpointManager<ParticipantData_t> {
                            bool historic_sample = false,
                            DCPS::MessageId id = DCPS::SAMPLE_DATA);
 
+    void write_control_msg(DCPS::Message_Block_Ptr payload,
+      size_t size,
+      DCPS::MessageId id,
+      DCPS::SequenceNumber seq = DCPS::SequenceNumber());
+
   private:
     DCPS::SequenceNumber seq_;
   };
 
-  class SedpWriter : public Writer {
+  class SecurityWriter : public Writer {
```

Comment:
I've looked into that earlier, but it seemed less beneficial than writer refactoring. `assoc()`, `data_received()`, and `remove_associations`  probably belong to the base Reader class. `notify_subscription_disconnected`, `notify_subscription_reconnected`, `notify_subscription_lost`, (and `remove_associations`) are empty in sedp. If we decide going deeper, refactoring may impact TransportReceiveListener and DataReaderImpl, which would be more invasive than it was for the writer. So I was not sure if was worth the effort, but if it is I'd like to have a design discussion first.

Reply:
I agree with the points you've made here.  However I think if we look deeper at what the current Reader does, `Sedp::Reader::data_received` has a large if/else-if chain to determine what type of writer sent the data.  If we split up the Reader, this wouldn't be needed.  Each type of Reader would either know type to expect, or have a small list (2 or 3) options to select from.

So the larger point, does this belong with this PR?  I think it does since this PR is doing the same split on the Writer side.  It will help readability/maintainability going forward.